backend/src/main/java/ai/verta/modeldb/entities/DatasetEntity.java

```diff
@@ -216,21 +226,30 @@ public void setDeleted(Boolean deleted) {
     this.deleted = deleted;
   }
 
-  public Dataset getProtoObject() throws InvalidProtocolBufferException {
-    return Dataset.newBuilder()
-        .setId(getId())
-        .setName(getName())
-        .setOwner(getOwner())
-        .setDescription(getDescription())
-        .addAllTags(RdbmsUtils.convertTagsMappingListFromTagList(getTags()))
-        .setDatasetVisibilityValue(getDataset_visibility())
-        .setDatasetTypeValue(getDataset_type())
-        .addAllAttributes(
-            RdbmsUtils.convertAttributeEntityListFromAttributes(getAttributeMapping()))
-        .setTimeCreated(getTime_created())
-        .setTimeUpdated(getTime_updated())
-        .setWorkspaceId(getWorkspace())
-        .setWorkspaceTypeValue(getWorkspace_type())
-        .build();
+  public void setCreated(Boolean created) {
+    this.created = created;
+  }
+
+  public Dataset getProtoObject(RoleService roleService) throws InvalidProtocolBufferException {
+    Dataset.Builder datasetBuilder =
+        Dataset.newBuilder()
+            .setId(getId())
+            .setName(getName())
+            .setOwner(getOwner())
+            .setDescription(getDescription())
+            .addAllTags(RdbmsUtils.convertTagsMappingListFromTagList(getTags()))
+            .setDatasetTypeValue(getDataset_type())
+            .addAllAttributes(
+                RdbmsUtils.convertAttributeEntityListFromAttributes(getAttributeMapping()))
+            .setTimeCreated(getTime_created())
+            .setTimeUpdated(getTime_updated())
+            .setWorkspaceId(getWorkspace())
+            .setWorkspaceTypeValue(getWorkspace_type());
+
+    GetResourcesResponseItem repositoryResource =
+        roleService.getEntityResource(
+            String.valueOf(this.id), ModelDBResourceEnum.ModelDBServiceResourceTypes.DATASET);
+    datasetBuilder.setVisibility(repositoryResource.getVisibility());
```

Comment:
Owner, visibility, workspace and custom permissions all must come from UAC and _not_ the local db. It should be safe for me to drop the columns from the local db that have been migrated.

Reply:
Currently, visibility comes from UAC (line 250) but the owner and workspace come from the local DB. i will update it as well.

Reply:
Updated owner and workspace as well now.